drivers/storage/s3fs/storage/s3fs_storage.go

```diff
@@ -0,0 +1,392 @@
+
+	iid := context.MustInstanceID(ctx)
+	return &types.Instance{
+		Name:         iid.ID,
+		InstanceID:   iid,
+		ProviderName: iid.Driver,
+	}, nil
+}
+
+// Volumes returns all volumes or a filtered list of volumes.
+func (d *driver) Volumes(
+	ctx types.Context,
+	opts *types.VolumesOpts) ([]*types.Volume, error) {
+
+	vols, err := d.toTypeVolumes(ctx, opts.Attachments)
+	if err != nil {
+		return nil, goof.WithError("error getting s3fs volumes", err)
+	}
+	return vols, nil
+}
+
+// VolumeInspect inspects a single volume.
+func (d *driver) VolumeInspect(
+	ctx types.Context,
+	volumeID string,
+	opts *types.VolumeInspectOpts) (*types.Volume, error) {
+
+	return d.getVolume(ctx, volumeID, opts.Attachments)
+}
+
+// VolumeCreate creates a new volume.
+func (d *driver) VolumeCreate(ctx types.Context, volumeName string,
+	opts *types.VolumeCreateOpts) (*types.Volume, error) {
+
+	if opts.Encrypted != nil && *opts.Encrypted {
+		return nil, types.ErrNotImplemented
+	}
+
+	_, err := d.svc.CreateBucket(
+		&awss3.CreateBucketInput{Bucket: &volumeName})
+	if err != nil {
+		ctx.WithField(
+			"volumeName", volumeName).WithError(err).Error(
+			"error creating s3 bucket")
+		return nil, goof.WithFieldE(
+			"volumeName", volumeName, "error creating s3 bucket", err)
+	}
+
+	return d.toTypeVolume(ctx, volumeName, types.VolAttNone), nil
+}
+
+// VolumeCreateFromSnapshot creates a new volume from an existing snapshot.
+func (d *driver) VolumeCreateFromSnapshot(
+	ctx types.Context,
+	snapshotID, volumeName string,
+	opts *types.VolumeCreateOpts) (*types.Volume, error) {
+	// TODO Snapshots are not implemented yet
+	return nil, types.ErrNotImplemented
+}
+
+// VolumeCopy copies an existing volume.
+func (d *driver) VolumeCopy(
+	ctx types.Context,
+	volumeID, volumeName string,
+	opts types.Store) (*types.Volume, error) {
+	// TODO Snapshots are not implemented yet
+	return nil, types.ErrNotImplemented
+}
+
+// VolumeSnapshot snapshots a volume.
+func (d *driver) VolumeSnapshot(
+	ctx types.Context,
+	volumeID, snapshotName string,
+	opts types.Store) (*types.Snapshot, error) {
+	// TODO Snapshots are not implemented yet
+	return nil, types.ErrNotImplemented
+}
+
+// VolumeRemove removes a volume.
+func (d *driver) VolumeRemove(
+	ctx types.Context,
+	volumeID string,
+	opts *types.VolumeRemoveOpts) error {
+
+	_, err := d.getVolume(ctx, volumeID, types.VolAttNone)
+	if err != nil {
+		ctx.WithField("volumeID", volumeID).WithError(err).Error(
+			"error finding bucket to delete")
+		return err
+	}
+
+	if _, err = d.svc.DeleteBucket(
+		&awss3.DeleteBucketInput{Bucket: &volumeID}); err != nil {
+		ctx.WithField(
+			"volumeID", volumeID).WithError(err).Error(
+			"error deleting s3 bucket")
+		return goof.WithFieldE(
+			"volumeID", volumeID, "error deleting s3 bucket", err)
+	}
+
+	return nil
+}
+
+// VolumeAttach attaches a volume and provides a token clients can use
+// to validate that device has appeared locally.
+func (d *driver) VolumeAttach(
```

Comment:
Can we add  a comment that for s3fs, the attach is not performed server-side, but is left to the executor?